File: solutions/prolog/resistor-color/1/resistor_color.pl
% Shree DR.MDD ke charno me samarpit
:- module(resistor_color, [color_code/2, colors/1]).

color_code(Tone, Rank) :-
    shade_set(Deck),
    nth0(Rank, Deck, Tone), !.

colors(Deck) :-
    shade_set(Deck).

shade_set([
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white"
]).

File: solutions/elixir/resistor-color/1/lib/resistor_color.ex
defmodule ResistorColor do
  @moduledoc false

  @colors [
    :black,
    :brown,
    :red,
    :orange,
    :yellow,
    :green,
    :blue,
    :violet,
    :grey,
    :white
  ]

  @spec code(String.t()) :: non_neg_integer
  def code(color) do
    Enum.find_index(@colors, &(&1 == color))
  end
end